sycl/include/sycl/backend.hpp

```diff
@@ -328,7 +328,7 @@ make_context(
     const async_handler &Handler = {}) {
   return detail::make_context(
       detail::ur::cast<ur_native_handle_t>(BackendObject), Handler, Backend,
-      false /* KeepOwnership */);
+      true /* KeepOwnership */);
```

Comment:
It seems like this changes the behavior for all paths, i.e. also L0, HIP and CUDA. Is that safe?

Reply:
I'll look into this.